plugins/inputs/aliyuncms/README.md

```diff
@@ -0,0 +1,142 @@
+2. AccessKey STS token credential if `access_key_id`, `access_key_secret`, `access_key_sts_token` is specified
+3. AccessKey credential if `access_key_id`, `access_key_secret` is specified
+4. Ecs Ram Role Credential if `role_name` is specified
+5. RSA keypair credential if `private_key`, `public_key_id` is specified
+6. Environment variables credential
+7. Instance metadata credential
+
+### Configuration:
+
+```toml
+  ## Aliyun Credentials
+  ## Credentials are loaded in the following order
+  ## 1) Ram RoleArn credential
+  ## 2) AccessKey STS token credential
+  ## 3) AccessKey credential
+  ## 4) Ecs Ram Role credential
+  ## 5) RSA keypair credential
+  ## 6) Environment variables credential
+  ## 7) Instance metadata credential
+  
+  # access_key_id = ""
+  # access_key_secret = ""
+  # access_key_sts_token = ""
+  # role_arn = ""
+  # role_session_name = ""
+  # private_key = ""
+  # public_key_id = ""
+  # role_name = ""
+  
+  # The minimum period for AliyunCMS metrics is 1 minute (60s). However not all
+  # metrics are made available to the 1 minute period. Some are collected at
+  # 3 minute, 5 minute, or larger intervals.
+  # See: https://help.aliyun.com/document_detail/51936.html?spm=a2c4g.11186623.2.18.2bc1750eeOw1Pv
+  # Note that if a period is configured that is smaller than the minimum for a
+  # particular metric, that metric will not be returned by the Aliyun OpenAPI
+  # and will not be collected by Telegraf.
+  #
+  ## Requested AliyunCMS aggregation Period (required - must be a multiple of 60s)
+  period = "5m"
+  
+  ## Collection Delay (required - must account for metrics availability via AliyunCMS API)
+  delay = "1m"
+  
+  ## Recommended: use metric 'interval' that is a multiple of 'period' to avoid
+  ## gaps or overlap in pulled data
+  interval = "5m"
+  
+  ## Metric Statistic Project (required)
+  project = "acs_slb_dashboard"
+  
+  ## Maximum requests per second, default value is 200
+  ratelimit = 200
+  
+  ## Discovery regions set the scope for object discovery, the discovered info can be used to enrich
+  ## the metrics with objects attributes/tags. Discovery is supported not for all projects (if not supported, then 
+  ## it will be reported on the start - foo example for 'acs_cdn' project:
+  ## 'E! [inputs.aliyuncms] Discovery tool is not activated: no discovery support for project "acs_cdn"' )
+  ## Currently, discovery supported for the following projects:
+  ## - acs_ecs_dashboard
+  ## - acs_rds_dashboard
+  ## - acs_slb_dashboard
+  ## - acs_vpc_eip
+  ##
+  ## If not set, all regions would be covered, it can provide a significant load on API, so the recommendation here
+  ## is to limit the list as much as possible. Allowed values: https://www.alibabacloud.com/help/zh/doc-detail/40654.htm
+  discovery_regions = ["cn-hongkong"]
+  
+  ## how often the discovery API call executed (default 1m)
+  #discovery_interval = "1m"
```

Comment:
Why have discovery_interval if there's already an interval?

Reply:
The reason is that the nature of the API calls is very different.
 `interval` defines how often we ask API about ali-cloud object metrics. `discovery_interval` defines how often we ask API about what ali-cloud objects exist and what attributes they have. There is no need to ask too often about new objects as once they are created we eventually can get the info about them and enrich metrics with objects attributes.

To sum up: different intervals is to decreasing API calls frequency (we are rate-limitied) and the overhead it produce, especially in the case when we ask across all regions (`discovery_regions` attribute set to empty list).